client/src/hooks/useFetch.js

```diff
@@ -1,19 +1,25 @@
 import React from 'react';
 
 function useFetch(url) {
-  const [user, setUser] = React.useState(null);
+  const [data, setData] = React.useState([]);
+  const auth = localStorage.getItem('token');
```

Comment:
You are reading from localStorage on every render.
It is only required on fetch.
I would move this line into your `useEffect`.